### PR TITLE
Adding better Slack integration

### PR DESCRIPTION
**Pull request by gabrielboliveira**
Related to #40, #42. This adds a better implementation of Slack notifications, now you can create several integrations and mark all you want to notify on Servers.

### CODE REVIEW

**Comment by gabrielboliveira** [comment]
@dbpolito @wesleyhf @viniciusld

File: app/Http/Controllers/ServerController.php [hunk 1]
```diff
@@ -30,19 +32,24 @@ public function store(ServerRequest $request)
...
+        $integrations = SlackIntegration::all()->lists('name', 'id');
+
+        return view('server.form', compact('server', 'integrations'));
     }
 
     public function update(ServerRequest $request, Server $server)
     {
-        $server->fill($request->all());
-        $server->save();
+        $server->save($request->all());
```
Comment (dbpolito) [inline app/Http/Controllers/ServerController.php]: I think this wont work, you need to change to `$server->update`

File: app/Http/Controllers/SlackIntegrationController.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    public function edit(SlackIntegration $integration)
+    {
+        return view('integration.form', compact('integration'));
+    }
+
+    public function update(SlackIntegrationRequest $request, SlackIntegration $integration)
+    {
+        $integration->fill($request->all());
```
Comment (dbpolito) [inline app/Http/Controllers/SlackIntegrationController.php]: `$integration->update`

File: app/Http/Controllers/SlackIntegrationController.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    public function create()
+    {
+        $integration = new SlackIntegration();
+        return view('integration.form', compact('integration'));
+    }
+
+    public function store(SlackIntegrationRequest $request)
+    {
+        $integration = new SlackIntegration($request->all());
```
Comment (dbpolito) [inline app/Http/Controllers/SlackIntegrationController.php]: make this a one line `SlackIntegration::create`

File: app/Http/Controllers/ServerController.php [hunk 4]
```diff
@@ -18,7 +19,8 @@ public function index()
     public function create()
     {
         $server = new Server();
-        return view('server.form', compact('server'));
+        $integrations = SlackIntegration::all()->lists('name', 'id');
```
Comment (wesleyhf) [inline app/Http/Controllers/ServerController.php]: remove `all()` function, use `lists()` directly

File: app/Libraries/SlackLibrary.php [hunk 5]
```diff
@@ -1,34 +1,32 @@
...
-        $data = json_encode(array(
-            'channel'    => '#' . env('SLACK_CHANNEL', 'general'),
-            'username'   => env('SLACK_BOTNAME', 'deployer'),
-            'text'       => $this->message,
-            'icon_emoji' => env('SLACK_ICON', ':rocket:')
-        ));
+        $data = [];
+        if (array_get($params, 'icon')) {
+            $data['icon_emoji'] = array_get($params, 'icon');
+        }
```
Comment (wesleyhf) [inline app/Libraries/SlackLibrary.php]: You can do something like this:

````php
if ($icon = array_get($params, 'icon')) {
    $data['icon_emoji'] = $icon;
}
````

File: app/Libraries/SlackLibrary.php [hunk 6]
```diff
@@ -1,34 +1,32 @@
...
-        $ch = curl_init(env('SLACK_WEBHOOK'));
-        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");
-        curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        $result = curl_exec($ch);
-        curl_close($ch);
+        $client = new Client();
+        $result = $client->post(array_get($params, 'webhook'), [
+            'json' => $data
+        ]);
```
Comment (wesleyhf) [inline app/Libraries/SlackLibrary.php]: use `with()` method to save one line:

````php
$result = with(new Client)->post(array_get($params, 'webhook'), ['json' => $data]);
````

**Pull request merged**